### PR TITLE
Edit page components integration fixing

### PR DESCRIPTION
**Pull request by Ashima-Arora**
@micahmills 

Here we have added dt-tags-related fixes in global-functions.php
and removed extra function in jQuery code which checked duplicate in contact_email and contact_phone fields. Kindly look in the changes and provide us with your feedback. Thanks!

@gp-birender  @vijenderrr

### CODE REVIEW

**Comment by Ashima-Arora** [comment]
@micahmills 

We have noticed that there are few errors when checks runs even after we have rebased our branch. Kindly assist us. Thanks!

**Review by micahmills — CHANGES_REQUESTED** [review]
This is working great for location fields when the mapbox key is activated.

It doesn't appear to be working if there is no mapbox key and it fallback to the regular location field. 
Screenshot 2024-06-03 at 10 48 58 AM

Please check that we aren't breaking the standard location field with these changes.

**Comment by Ashima-Arora** [comment]
@micahmills 

After the discussions in our last call, we have put the **dt-location**  work on hold. 
We added ` ( $field_type === 'location_meta' )` to get the location field displayed. However, that wasn't worked on. We will resume work after you provide us with instructions on the **location field updates**. Thus, I will revert [these ](
https://github.com/DiscipleTools/disciple-tools-theme/pull/2510/files/ed8f8ab9bb58868ebc9d2f42bb0f11e9cc396b21#diff-a6cf6ac104a9f97ffc7db281c1858721c10487200efbbc76cd4acfe74af75a96R657) changes for now. Thank you!

File: dt-core/global-functions.php [hunk 1]
```diff
@@ -694,10 +695,7 @@ function render_field_for_display( $field_key, $fields, $post, $show_extra_contr
                     </dt-single-select>
             <?php elseif ( $field_type === 'tags' ): ?>
                 <?php $value = array_map(function ( $value ) {
-                    return [
-                        'id' => $value,
-                        'label' => $value,
-                    ];
+                    return $value;
```
Comment (Ashima-Arora) [inline dt-core/global-functions.php:698]: @micahmills,

 This code is related to the dt-tag field on the edit detail page. We made this change to make the DT-tag field visible on frontend(UI) also user is able to edit the same. 
Related code in web components is in this PR also
https://github.com/DiscipleTools/disciple-tools-web-components/pull/76#issuecomment-2182218619

We have made changes on both sides (DT-Theme as well as Web components)  are related to this dt-tags fix. 

Kindly check and let us know your feedback. Thank you!

cc: @gp-birender @vijenderrr

Comment (micahmills) [reply dt-core/global-functions.php:698]: Thank you @Ashima-Arora it was my mistake I thought I was testing the Web component PR with this branch, but I was not. I will merge this in and we can deal with the location_meta stuff at a later time.

**Review by micahmills — APPROVED** [review]

**Pull request merged**